contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -152,14 +157,16 @@ abstract contract PoSValidatorManager is
         if (weightToValueFactor == 0) {
             revert ZeroWeightToValueFactor();
         }
-
-        $._minimumStakeAmount = minimumStakeAmount;
-        $._maximumStakeAmount = maximumStakeAmount;
-        $._minimumStakeDuration = minimumStakeDuration;
-        $._minimumDelegationFeeBips = minimumDelegationFeeBips;
-        $._maximumStakeMultiplier = maximumStakeMultiplier;
-        $._weightToValueFactor = weightToValueFactor;
-        $._rewardCalculator = rewardCalculator;
+        // Efficient Assignment
+        $.config = PoSValidatorManagerConfig({
+            _minimumStakeAmount: minimumStakeAmount,
+            _maximumStakeAmount: maximumStakeAmount,
+            _minimumStakeDuration: minimumStakeDuration,
+            _minimumDelegationFeeBips: minimumDelegationFeeBips,
+            _maximumStakeMultiplier: maximumStakeMultiplier,
+            _weightToValueFactor: weightToValueFactor,
+            _rewardCalculator: rewardCalculator
+        });
```

Comment:
I don't think the gas savings here are worth the added code complexity and readability. Let's stick with keeping `PoSValidatorManagerStorage` as is.